agent/src/testflinger_agent/agent.py

```diff
@@ -417,6 +415,13 @@ def process_jobs(self):
                 results_basedir = self.client.config.get("results_basedir")
                 shutil.move(rundir, results_basedir)
 
+            # Complete cleanup only if server is reachable
+            if not self.client.is_server_reachable():
+                self.client.wait_for_server_connectivity()
```

Comment:
The `wait_for_server_connectivity` alone would be able to check this, right?

```suggestion
            self.client.wait_for_server_connectivity()
```